src/mod_muc_commands.erl

```diff
@@ -173,6 +182,39 @@ kick_user_from_room(Host, Name, Nick) ->
 %% Ancillary
 %%--------------------------------------------------------------------
 
+-spec verify_room(jid:server(), mod_muc:room(), jid:literal_jid()) ->
+    ok | {error, internal | not_found, term()}.
+verify_room(Host, RoomName, User) ->
+    MUCHost = gen_mod:get_module_opt_subhost(Host, mod_muc, mod_muc:default_host()),
+    BareRoomJID = jid:make(RoomName, MUCHost, <<"">>),
+    UserJID = jid:binary_to_bare(User),
+    verify_room(BareRoomJID, UserJID).
+
+-spec verify_room(jid:jid(), jid:jid()) ->
+    ok | {error, internal | not_found, term()}.
+verify_room(BareRoomJID, OwnerJID) ->
+    case mod_muc_room:can_access_room(BareRoomJID, OwnerJID) of
+        {ok, true} ->
+            ok;
+        {ok, false} ->
+            {error, internal, "room is locked"};
+        {error, not_found} ->
+            {error, not_found, "room does not exist"}
+    end.
+
+prep_jid(Sender, Recipient) ->
+    case {prep_jid(Sender), prep_jid(Recipient)} of
+        {{error, Msg}, _} -> {error, type_error, Msg};
+        {_, {error, Msg}} -> {error, type_error, Msg};
+        {S, R} -> {ok, S, R}
+    end.
+
+prep_jid(Jid) ->
+    case jid:binary_to_bare(Jid) of
+        error -> {error, io_lib:format("Invalid jid: ~p", [Jid])};
+        B -> B
+    end.
+
```

Comment:
Similar comments to those I had in `mod_commands`, and actually I'm wondering if the distinction between `binary_to_bare` and `from_binary` is really necessary, this could could be unified. But perhaps an empty resource in the jid is required, I don't know. If it is required, maybe in can be cleaned on the place where it is used and this code could be unified in some helper module with the code from the other module.